### PR TITLE
refactor(lint): Replace tslint with eslint

### LINKED ISSUE #4: [CHANGE] Need to replace ts-lint with es-lint
## Who will this change affect?

This change will affect the developers of the project

## Why this change?

TSLint which has been used for linting the code of this project is deprecated and no longer supported. The support for TypeScript linting has now shifted towards ESLint which is currently in full active development.

## What will this change accomplish?

This change will help us replace TSLint, which is deprecated in favor of ESLint which is currently actively supporting TypeScript linting.

## Useful links

* https://www.darraghoriordan.com/2020/03/06/upgrade-tslint-deprecated-to-eslint/#:~:text=Ts%20lint%20is%20deprecated%20now,continuous%20updates%20from%20the%20team.
* https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration

### PR DESCRIPTION
**Pull request by shubham-cpp**
## Description

Since tslint has been deprecated in favor of eslint. I've also included few more plugins(react-jsx-ally,etc) to make it more strict. Also added nabla eslint to vite so we can get error reporting in terminal itself. There are few more rules that I would like to discuss before moving forward
## Related Issue

Fixes #4 

## Motivation and Context

Favoring Eslint since its in active development.
Tools used:
- https://github.com/typescript-eslint/tslint-to-eslint-config
- https://medium.com/palantir/tslint-in-2019-1a144c2317a9
## How Has This Been Tested?

Ran `npm run dev` and everything works as expected

### CODE REVIEW

**Comment by shubham-cpp** [comment]
I would also highly recommend using [pnpm](https://pnpm.io/installation) instead of yarn since its much faster and better compatible with npm

**Comment by azrael3199** [comment]
> I would also highly recommend using [pnpm](https://pnpm.io/installation) instead of yarn since its much faster and better compatible with npm

Let's create an issue for this enhancement with the label `tools`

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,5159 @@
+lockfileVersion: 5.4
```
Comment (azrael3199) [inline pnpm-lock.yaml:1]: I guess we should delete this for now and add it in another PR when we add pnpm

**Review by azrael3199 — COMMENTED** [review]
I think we need to take the pnpm file out for now and add it later in another PR with an issue to it

**Review by azrael3199 — APPROVED** [review]
LGTM

**Pull request merged**